src/Blef.GameLogic/PokerHands/PokerHand.cs

```diff
@@ -1,57 +1,23 @@
-﻿using System;
-
-namespace Blef.GameLogic.PokerHands
+﻿namespace Blef.GameLogic.PokerHands
 {
     public abstract class PokerHand
     {
         public abstract bool IsOnTable(Table table);
 
-        protected abstract int Value
-        {
-            get;
-        }
-
-        public bool IsStrongerThan(PokerHand otherPokerHand)
+        public int CompareWith(PokerHand otherPokerHand)
         {
-            return this.Value > otherPokerHand.Value;
-        }
+            int genericValueResult = GenericPokerHand - otherPokerHand.GenericPokerHand;
 
-        protected int GetRankValue(Rank rank)
-        {
-            switch (rank)
+            if (genericValueResult == 0)
             {
-                case Rank.Nine:
-                    return 1;
-                case Rank.Ten:
-                    return 2;
-                case Rank.Jack:
-                    return 3;
-                case Rank.Queen:
-                    return 4;
-                case Rank.King:
-                    return 5;
-                case Rank.Ace:
-                    return 6;
-                default:
-                    throw new ArgumentOutOfRangeException(nameof(rank), rank, null);
+                return CompareWithinSameGenericPokerHand(otherPokerHand);
             }
-        }
 
-        protected int GetSuitValue(Suit suit)
-        {
-            switch (suit)
-            {
-                case Suit.Spades:
-                    return 4;
-                case Suit.Hearts:
-                    return 3;
-                case Suit.Diamonds:
-                    return 2;
-                case Suit.Clubs:
-                    return 1;
-                default:
-                    throw new ArgumentOutOfRangeException(nameof(suit), suit, null);
-            }
+            return genericValueResult;
         }
+
+        protected abstract int GenericPokerHand { get; }
```

Comment:
I agree it is not the best name, I am looking for better, looking at the detailed description how would you name it?